.github/workflows/code-quality-gates.yml

```
@@ -0,0 +1,395 @@
+        uses: actions/upload-artifact@v4
+        with:
+          name: performance-analysis
+          path: bundle-analysis.md
+          retention-days: 30
+
+  # Quality gate enforcement
+  quality-gate:
+    name: 🚪 Quality Gate
+    runs-on: ubuntu-latest
+    needs: [quality-analysis, security-analysis, performance-analysis]
+    if: always() && needs.detect-changes.outputs.should-run-quality == 'true'
```

actionlint (1.7.7)

property "detect-changes" is not defined in object type {performance-analysis: {outputs: {}; result: string}; quality-analysis: {outputs: {complexity-passed: string; coverage-passed: string; quality-score: string}; result: string}; security-analysis: {outputs: {}; result: string}}

(expression)